packages/swap/src/routes/v2/swap.ts

```diff
@@ -72,37 +74,39 @@ router.get(
 
     const originalInputAmount = swapRequest?.swapInputAmount;
 
-    const swaps = swapRequest?.swaps;
-
-    const rolledSwaps = swapRequest?.swaps.reduce(
-      (acc, curr) => {
-        if (curr.swapExecutedAt) {
-          acc.swappedOutputAmount = acc.swappedOutputAmount.plus(curr.swapOutputAmount ?? 0);
-          acc.swappedIntermediateAmount = acc.swappedIntermediateAmount.plus(
-            curr.intermediateAmount ?? 0,
-          );
-          acc.swappedInputAmount = acc.swappedInputAmount.plus(curr.swapInputAmount);
-          acc.lastExecutedChunk = curr;
-          acc.executedChunks += 1;
-          acc.fees = acc.fees.concat(...curr.fees);
-        } else {
-          acc.currentChunk = curr;
-        }
-        return acc;
-      },
-      {
-        swappedOutputAmount: new Prisma.Decimal(0),
-        swappedIntermediateAmount: new Prisma.Decimal(0),
-        swappedInputAmount: new Prisma.Decimal(0),
-        executedChunks: 0,
-        currentChunk: null as null | NonNullable<typeof swaps>[number],
-        lastExecutedChunk: null as null | NonNullable<typeof swaps>[number],
-        isDca: Boolean(
-          swapDepositChannel?.chunkIntervalBlocks && swapDepositChannel.chunkIntervalBlocks > 1,
-        ),
-        fees: [] as SwapFee[],
-      },
-    );
+    const swaps = swapRequest?.swaps.filter((swap) => swap.type !== 'GAS');
+
+    const rolledSwaps = swaps?.length
+      ? swaps.reduce(
```

Comment:
```suggestion
    const rolledSwaps = swaps?.reduce(
```

Reply:
we don't want to return the 0s from the default value in the status response (check line 203). At least that felt most natural to me.

Reply:
We just had an ingress -> gas swap (for the egress fee) -> egress . No principal or other swaps done.

Reply:
yeah but the optional chain will have the same result as this ternary

Reply:
hmm, not when the array of swaps is empty though? rolledSwaps will be initialized with the default value of the reducer